### PR TITLE
Release v1.2.5

### PR DESCRIPTION
**Pull request by jonathan-dilorenzo**
Also updated changelog and added a new section for the next release.

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -8619,14 +8619,21 @@ The P4 compiler should provide:
...
 - Errors when annotations are used incorrectly (e.g., an annotation
   expecting a parameter but used without arguments, or with arguments
-  of the wrong type
+  of the wrong type)
 - Warnings for unknown annotations.
 
 # Appendix: Revision History { #sec-revision-history; @h1:"A" }
 
-## Summary of changes made in version 1.2.4
+## Summary of changes made in unreleased version
```
Comment (rcgoodfellow) [inline p4-16/spec/P4-16-spec.mdk:8627]: What's the intention of this section?

Comment (jonathan-dilorenzo) [reply p4-16/spec/P4-16-spec.mdk:8627]: The idea will be for PRs that make changes to already add bullet for what they changed. Then, ideally, cutting a new release will be as easy as changing "unreleased version" to "version X.Y.Z"

**Review by rcgoodfellow — APPROVED** [review]

**Comment by jonathan-dilorenzo** [comment]
@jafingerhut and @jnfoster, since neither Ryan or I have done this before, I'd like approval from you too before merging (though I'll update the date I say it's released to the date we actually do the release once I've got the approvals).

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -8619,14 +8619,21 @@ The P4 compiler should provide:
...
 
+## Summary of changes made in version 1.2.5, released October 10, 2024
```
Comment (jonathan-dilorenzo) [inline p4-16/spec/P4-16-spec.mdk:8629]: Reminder to self to update this to the true date once known.

**Review by jafingerhut — APPROVED** [review]
Looks good to me.

**Pull request merged**